src/ngx_http_tnt_module.c

```diff
@@ -1616,7 +1616,7 @@ ngx_http_tnt_urldecode(ngx_http_request_t *r, ngx_str_t *src)
 
         c = src->data[s++];
 
-        if (c == '%' && (ngx_uint_t) (s + 2) < src->len) {
+        if (c == '%' && (ngx_uint_t) (s + 2) <= src->len) {
```

Comment:
It would be good to provide an example of incorrectly parsed URL and add a test case (or mention existing one that reveals the problem).

Reply:
Sorry, I pointed the wrong issue. See the new commit message, I just restored old commit with description (in #120) and test.

Reply:
Now I see. Revertion of one fix was mistakely land inside another fix.